parcels/field.py

```diff
@@ -1545,18 +1553,76 @@ def __init__(self, filename, dimensions, indices, netcdf_engine, timestamp=None,
         self.interp_method = interp_method
         self.data_full_zdim = data_full_zdim
         self.field_chunksize = field_chunksize
+        self.chunk_mapping = None
+        self.rechunk_callback_fields = rechunk_callback_fields
 
     def __enter__(self):
         if self.netcdf_engine == 'xarray':
             self.dataset = self.filename
             return self
+
+        # ==== check-opening ==== #
+        try:
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=True, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = True
+        except:
+            logger.warning_once("File %s could not be decoded properly by xarray (version %s).\n         It will be opened with no decoding. Filling values might be wrongly parsed."
+                                % (self.filename, xr.__version__))
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=False, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = False
+
+        init_chunk_dict = {}
+        if isinstance(self.field_chunksize, dict):
+            init_chunk_dict = self.field_chunksize
+        elif isinstance(self.field_chunksize, tuple):
+            spatStart = 0
+            if 'time' in self.dimensions and self.dimensions['time'] in self.dataset.dims:
+                if(len(self.dimensions) > 2):
+                    init_chunk_dict[self.dimensions['time']] = self.field_chunksize[0]
+                    spatStart += 1
+                else:
+                    init_chunk_dict[self.dimensions['time']] = 1
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims - Please specify field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['depth']))
+            if 'depth' in self.dimensions and self.dimensions['depth'] in self.dataset.dims:
+                if(len(self.dimensions) > 3):
+                    init_chunk_dict[self.dimensions['depth']] = self.field_chunksize[1]
+                    spatStart += 1
+                else:
+                    init_chunk_dict[self.dimensions['depth']] = 1
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims - Please specify field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['depth']))
```

Comment:
Make this a function/method so that we can more easily repeat it for each dimension?

Reply:
agreed

Reply:
I made the error message now being generated by a function. Unfortunately, reworking the whole code section into a simple loop doesn't work because indices for time or depth can be optional, but when actually specifying a `field_chunksize`, there should be at least values for *lat* and *lon* (or: their equivalents). Hence, those dimensions are actually treated differently.

Reply:
I also replaced `raise Warning(...)` with `logger.warning_once`, because apparently, raising a warning results in pytest failure. Learning experience +1